File: selenium/dotnet/OtpAuthenticatorTests.cs
/**
 * IMPORTANT: This test uses Mailosaur Authenticator, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/devices)
 */

using System;
using Xunit;
using Mailosaur;

namespace mailosaur_example;

public class OtpAuthenticatorTests()
{
    [Fact (Skip = "Reason")]
    public void GenerateOneTimePasscode()
    {
        DotNetEnv.Env.TraversePath().Load();

        // Instantiate Mailosaur client (reads MAILOSAUR_API_KEY from environment)
        var mailosaur = new MailosaurClient();

        /**
         * This is a base32-encoded shared secret.
         * Typically this is the value shown to a user if they cannot scan an on-screen QR code.
         * Learn more at https://mailosaur.com/docs/mfa
        */
        var sharedSecret = "ONSWG4TFOQYTEMY=";
        var passcode = mailosaur.Devices.Otp(sharedSecret);

        Console.WriteLine($"\nAuthenticator otp code - {passcode.Code}"); // "564214"
    }
}

File: php/tests/OtpAuthenticatorTests.php
<?php

/**
 * IMPORTANT: This test uses Mailosaur Authenticator, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/devices)
 */

namespace MailosaurExample;

use Mailosaur\MailosaurClient;
use Mailosaur\Models\MessageSummary;
use Mailosaur\Models\SearchCriteria;

class OtpAuthenticatorTests extends \PHPUnit\Framework\TestCase
{
  /** @var \Mailosaur\MailosaurClient */
  protected static $client;
  
  public static function setUpBeforeClass(): void
  {
    $apiKey = $_ENV['MAILOSAUR_API_KEY'];
    
    // Instantiate Mailosaur client with api key
    self::$client = new MailosaurClient($apiKey);
  }
  
  /** @test */
  public function generateOneTimePasscode(): void
  {
    $this->markTestSkipped('Reason');

    /**
     * This is a base32-encoded shared secret.
     * Typically this is the value shown to a user if they cannot scan an on-screen QR code.
     * Learn more at https://mailosaur.com/docs/mfa
     */
    $sharedSecret = 'ONSWG4TFOQYTEMY=';
    $currentOtp = self::$client->devices->otp($sharedSecret);

    print(PHP_EOL . PHP_EOL . 'Authenticator otp code - ' . $currentOtp->code); // "564214"
  }
}